Storage/SQL/SQLiteBookmarksModel.swift

```diff
@@ -855,4 +890,10 @@ public class MergedSQLiteBookmarks: BookmarksModelFactorySource {
         self.local = SQLiteBookmarks(db: db)
         self.buffer = SQLiteBookmarkBufferStorage(db: db)
     }
+
+    public func getURLForKeywordSearch(keyword: String) -> Deferred<Maybe<String>> {
+        return modelFactory >>== {
+            return ($0 as! KeywordSearchSource).getURLForKeywordSearch(keyword)
```

Comment:
@rnewman I wish I could avoid downcasting and instead, declare something like:
public var modelFactory: protocol<Deferred<Maybe<BookmarksModelFactory>>, KeywordSearchSource>

Is that possible?